### PR TITLE
Mouse Class Interface

### LINKED ISSUE #11: Plan the Mouse class
image

### PR DESCRIPTION
**Pull request by blhough**
Let's agree on an initial interface for the Mouse class
Closes #11

### CODE REVIEW

File: Mouse.h [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    void runMaze();
+
+    // In millimeter per second.
+    void setMappingSpeed(unsigned mmps);
+    void setRunningSpeed(unsigned mmps);
+
+    void reset();
+
+  private:
+    void followPath(const Path &path);
```
Comment (klanmiko) [inline Mouse.h]: should we add a function void goToCell(CellCoordinate cell); that creates a path from the current cell to the cell using the path finder and calls followPath?

File: Mouse.h [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    void move(DirectionVector movement);
+
+    Maze maze;
+    Hardware bot;
+
+    Coordinate position;
+    Direction facing;
```
Comment (klanmiko) [inline Mouse.h]: Should we also create a private function to update these two variables by looking at the bot?

**Review by klanmiko — COMMENTED** [review]
Could use a few more functions

File: Mouse.h [hunk 3]
```diff
@@ -25,7 +25,7 @@ class Mouse {
     Maze maze;
     Hardware bot;
 
-    Coordinate position;
+    NodeCoordinate position;
```
Comment (klanmiko) [inline Mouse.h]: CellCoodinate. Because there is no way the robot can be in a wall

Comment (blhough) [reply Mouse.h]: If we want to allow diagonals then the mouse would need to be in between cells.
image

File: Mouse.h (hunk 1 above)
Comment (blhough) [reply Mouse.h]: Yes that sounds good. Can you think of anything else for the public interface?

**Review by klanmiko — APPROVED** [review]
I think the initial interface is fine for now.

In the future we should think about adding states to the Mouse class so that its logic can be self contained, and add a process function pointer that controls the state machine and handles logic, updates values.

This allows the simulate.main_loop() or loop() to remember less of the mouse state. Testing can also be done by mocking the process() function.

**Pull request merged**